frontends/common/resolveReferences/referenceMap.h

```diff
@@ -67,17 +68,17 @@ class ReferenceMap final : public ProgramMap, public NameGenerator, public Decla
     bool isv1;
 
     /// Maps paths in the program to declarations.
-    ordered_map<const IR::Path *, const IR::IDeclaration *> pathToDeclaration;
+    absl::flat_hash_map<const IR::Path *, const IR::IDeclaration *, Util::Hash> pathToDeclaration;
```

Comment:
Note: In general we should be quite careful when replacing `ordered_map` with something that does not keep order as `ordered_map` could be used to make sure the compiler is deterministic (and nondeterminism bugs are extremely hard to pinpoint :-/).

In this particular case though, I don't see any risk of nondeterminism as nothing iterates over the containers (expect possibly the debug print).

Reply:
I double checked all `ordered_map` differences when made this changes. In many cases there is no iteration at all besides the debug priting. So, just having stable debug output does not justify the generic slowdowns.